src/app/workbench/WorkbenchViews.tsx

```diff
@@ -22,127 +41,66 @@ export function WorkbenchViews({ index, type }: IWorkbenchViewsProps) {
   const { views, selection, commentsOpen, itemIDType } = ordino.workbenches[index];
   const [setRef] = useCommentPanel({ selection, itemIDType, commentsOpen, isFocused: type === EWorkbenchType.FOCUS });
 
-  let wb = null;
+  const [mosaicState, setMosaicState] = useState<any>(null);
+  const [mosaicViewCount, setMosaicViewCount] = useState<number>(1);
+
+  const [mosaicDrag, setMosaicDrag] = useState<boolean>(false);
+
+  React.useEffect(() => {
+    if (mosaicState === null) {
+      setMosaicState(views[0].uniqueId);
+      return;
+    }
+    if (views.length > mosaicViewCount) {
+      const path = getPathToCorner(mosaicState, Corner.TOP_RIGHT);
+      const parent = getNodeAtPath(mosaicState, dropRight(path)) as MosaicParent<string>;
+      const destination = getNodeAtPath(mosaicState, path) as MosaicNode<string>;
+      const direction: MosaicDirection = parent ? getOtherDirection(parent.direction) : 'row';
```

Comment:
```suggestion
      const direction: MosaicDirection = parent ? getOtherDirection(parent.direction) : 'row';
      const newViewId: string = views[views.length - 1].uniqueId; // assumes that the new view is appended to the array
```

Maybe extract to a variable and add a comment that new views are appended.

Reply:
Agreed, done. Should also mention here the way this works will probably not quite hold when we get provenance in, since it relies on views being added one at a time and in order. But can address that when the need arises?

Reply:
Good question! I think, we don't need a todo here, as we need to test everything again once the provenance is in place. Maybe you give @kirangadhave a heads up.